### ISSUE TITLE
ek-components createTranslator usage broken

### ISSUE THREAD

**Issue by dbnicholson** [endlessm/kolibri-explore-plugin#817]
It looks like #777 was good enough for building but not good enough for runtime. I tried running the dev server and it just spins on the loader with this in the console:

```
Uncaught TypeError: Cannot read properties of undefined (reading 'createTranslator')
    at ./packages/ek-components/src/constants.js (constants.js:1:1)
    at __webpack_require__ (bootstrap:19:1)
    at ./node_modules/kolibri-tools/node_modules/vue-loader/lib/index.js??vue-loader-options!./packages/ek-components/src/components/EkCard.vue?vue&type=script&lang=js& (EkCard.vue:40:1)
    at __webpack_require__ (bootstrap:19:1)
    at ./packages/ek-components/src/components/EkCard.vue?vue&type=script&lang=js& (EkButtonsBar.vue:1:1)
    at __webpack_require__ (bootstrap:19:1)
    at ./packages/ek-components/src/components/EkCard.vue (EkButtonsBar.vue:95:1)
    at __webpack_require__ (bootstrap:19:1)
    at ./packages/ek-components/src/main.js (constants.js:211:1)
    at __webpack_require__ (bootstrap:19:1)
```

That makes sense because `window.kolibri` is set dynamically by the frontend. In the console, `window.kolibri` is currently `undefined`. Somehow this will need to be delayed until it can be resolved. I don't know how to do that.

CC @pwithnall @manuq

**Comment by pwithnall** [comment]
Uff, this is my fault. I rushed too much on Friday and didn’t test the final iteration of my branch.

Possible solutions which come to mind:
 * Add some synchronisation between `main.js` and when it imports `constants.js`, but that’s hard (impossible?) because it re-exports `constants.js`.
 * Move `ekComponentsConstantsStrings` inside a function so that `window.kolibri.createTranslator` doesn’t have to be dereferenced until the first translated string is actually needed.

**Comment by dbnicholson** [comment]
> Uff, this is my fault. I rushed too much on Friday and didn’t test the final iteration of my branch.

Well, I was certainly pushing to get something in, so don't take all the blame. It would help if we even a minimal smoke test that ran.

**Comment by manuq** [comment]
On master I now see `TypeError: Cannot read properties of undefined (reading 'translators')`. This is for namespaces `EkComponentsConstants` and `EkComponentsUtils`. I understand that it happens because the ek-components is using `createTranslator` through `window.kolibri` and is not instantiating the `KolibriApi` class.

Also it would be better if the translators cache is reused. Here is a proposed solution https://github.com/endlessm/kolibri-explore-plugin/pull/826

But that doesn't work with our version of eslint. So here is an alternative solution: https://github.com/endlessm/kolibri-explore-plugin/pull/827

**Comment by dbnicholson** [comment]
I think #827 fixes this. I ran it locally and it seemed to work.

**Comment by dbnicholson** [comment]
This was released in 7.0.0 and is ready to test.

**Comment by erikos** [comment]
Spanish localization of the UI is fully working 7.1 onwards.
